src/core/compact_object.cc

```diff
@@ -208,6 +213,23 @@ pair<void*, bool> DefragSet(unsigned encoding, void* ptr, float ratio) {
   }
 }
 
+pair<void*, bool> DefragZSet(unsigned encoding, void* ptr, float ratio) {
+  switch (encoding) {
+    // Listpack is stored as a single contiguous array
+    case OBJ_ENCODING_LISTPACK: {
+      return DefragListPack((uint8_t*)ptr, ratio);
+    }
+
+    // SKIPLIST really means ScoreMap
+    case OBJ_ENCODING_SKIPLIST: {
+      return DefragSortedMap((detail::SortedMap*)ptr, ratio);
+    }
+
+    default:
+      ABSL_UNREACHABLE();
```

Comment:
this (may?) crash, and since it's easy to recover from this situation (just do nothing...), I'd add a `LOG(ERROR)` instead and remove the `ABSL_UNREACHABLE`

Reply:
If the underline type is not `SKIPLIST` or `LISTPACK` we are in big trouble and we are on path triggering UB at some point or the other. IMO it's better to crash here and we do the same for similar paths

Reply:
Not necessarily. For example, if this function is called with a non-zset object by mistake.

Reply:
I see other places in this file also use this, so please ignore. let's be consistent.